android/quest/src/main/java/org/smartregister/fhircore/quest/util/IterableUtil.kt

```diff
@@ -33,22 +33,18 @@ fun <A, B> nonNullGetOrDefault(map: Map<A, B>, key: A?, defaultValue: B): B {
 }
 
 /**
- * Function to convert the elements of an array that have paramType PARAM_DATA to a map of their
- * keys to values.
+ * Function to convert the elements of an array that have paramType [ActionParameterType.PARAMDATA]
+ * to a map of their keys to values. It also returns [emptyMap] if [actionParameters] is null.
  *
  * @property array The array of ActionParameter elements to convert
- *
  * @return Map of the values or emptyMap if [array] is null
  */
-fun convertActionParameterArrayToMap(array: Array<ActionParameter>?): Map<String, String> {
-  val paramsMap: MutableMap<String, String> = mutableMapOf()
-  if (array != null) {
-    for (param in array) {
-      if (param.paramType == ActionParameterType.PARAMDATA && !param.value.isNullOrEmpty()) {
-        paramsMap[param.key] = param.value
-      }
-    }
-    return paramsMap.toMap()
-  }
-  return emptyMap()
+fun convertActionParameterArrayToMap(
```

Comment:
Can we convert this to an extension of `Array<ActionParameter>?` ? and possibly move this to `ConfigExtensions` file?